http/src/main/resources/reference.conf

```diff
@@ -56,6 +56,66 @@ gce {
   gceReservedMemory = 1g
 }
 
+vpc {
+  highSecurityProjectNetworkLabel = "vpc-network-name"
+  highSecurityProjectSubnetworkLabel = "vpc-subnetwork-name"
+  networkName = "leonardo-network"
+  networkTag = "leonardo"
+  autoCreateSubnetworks = false
+  # note the below 3 fields are not used if autoCreateSubnetworks is true
+  subnetworkName = "leonardo-subnetwork"
+  subnetworkRegion = "us-central1"
+  subnetworkIpRange = "10.128.0.0/20"
+  firewallsToAdd = [
+    # allows Leonardo proxy traffic on port 443
+    {
+      name = "leonardo-allow-https"
+      network = ${vpc.networkName}
+      sourceRanges = ["0.0.0.0/0"]
+      allowed = [
+        {
+          protocol = "tcp"
+          port = "443"
+        }
+      ]
+    },
+    # allows dataproc nodes to communicate with each other via internal IP
+    {
+      name = "leonardo-allow-internal"
+      network = ${vpc.networkName}
+      sourceRanges = ["10.128.0.0/20"]
+      allowed = [
+        {
+          protocol = "tcp"
+          port = "0-65535"
+        },
+        {
+          protocol = "udp"
+          port = "0-65535"
+        },
+        {
+          protocol = "icmp"
+        }
+      ]
+    },
+    # allows SSH access from the Broad network or VPN
+    {
+      name = "leonardo-allow-broad-ssh"
+      network = ${vpc.networkName}
+      sourceRanges= ["69.173.112.0/21", "69.173.127.232/29", "69.173.127.128/26", "69.173.127.0/25", "69.173.127.240/28", "69.173.127.224/30", "69.173.127.230/31", "69.173.120.0/22", "69.173.127.228/32", "69.173.126.0/24", "69.173.96.0/20", "69.173.64.0/19", "69.173.127.192/27", "69.173.124.0/23"]
+      allowed = [
+        {
+          protocol = "tcp"
+          port = 22
+        }
+      ]
+    }
+  ]
+  firewallsToRemove = ["default-allow-rdp", "default-allow-ssh", "leonardo-notebooks-rule"]
```

Comment:
I also see `default-allow-icmp` on the [list](https://cloud.google.com/vpc/docs/firewalls#more_rules_default_vpc). Do we care about that?

Reply:
I am not sure, I guess that enables `ping`. I was going by [this doc](https://docs.google.com/document/d/1adV0LC2f_GIpl3A1AeoQuNiwcP59NToIt6VYT3xRCkU/edit) which doesn't mention icmp traffic so I'd just as soon leave it.